crates/hyperswitch_connectors/src/connectors/payload/transformers.rs

```diff
@@ -0,0 +1,643 @@
+        }
+    }
+}
+// PaymentsResponse - Updated to match Payload API status values
+#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum PayloadPaymentStatus {
+    #[serde(rename = "authorized")]
+    Authorized,
+    #[serde(rename = "processed")]
+    Processed,
+    #[serde(rename = "cancelled")]
+    Cancelled,
+    #[serde(rename = "voided")]
+    Voided,
+    #[serde(rename = "failed")]
+    Failed,
+    #[serde(rename = "pending")]
+    #[default]
+    Pending,
+}
+
+impl From<PayloadPaymentStatus> for common_enums::AttemptStatus {
+    fn from(item: PayloadPaymentStatus) -> Self {
+        match item {
+            PayloadPaymentStatus::Authorized => Self::Authorized,
+            PayloadPaymentStatus::Processed => Self::Charged,
+            PayloadPaymentStatus::Cancelled | PayloadPaymentStatus::Voided => Self::Voided,
+            PayloadPaymentStatus::Failed => Self::Failure,
+            PayloadPaymentStatus::Pending => Self::Pending,
+        }
+    }
+}
+
+// Response structure matching actual Payload API response
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct PayloadPaymentsResponse {
+    pub id: String,
+    #[serde(rename = "type")]
+    pub r#type: String,
+    pub amount: f64,    // Payload returns amount as float
+    pub status: String, // Payload returns status as string
+    pub status_code: Option<String>,
+    pub status_message: Option<String>,
+    pub created_at: String,
+    pub avs: Option<String>,
+    pub payment_method: Option<PayloadPaymentMethodResponse>,
+    pub payment_method_id: Option<String>,
+    pub processing_id: Option<String>,
+    pub ref_number: Option<String>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct PayloadPaymentMethodResponse {
+    id: String,
+    #[serde(rename = "type")]
+    r#type: String,
+    card: Option<PayloadCardResponse>,
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct PayloadCardResponse {
+    card_brand: String,
+    card_number: String, // Masked card number like "xxxxxxxxxxxx4242"
+    card_type: String,
+    expiry: String,
+}
+
+impl<F, T> TryFrom<ResponseRouterData<F, PayloadPaymentsResponse, T, PaymentsResponseData>>
+    for RouterData<F, T, PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: ResponseRouterData<F, PayloadPaymentsResponse, T, PaymentsResponseData>,
+    ) -> Result<Self, Self::Error> {
+        // Map Payload string status to AttemptStatus
+        let status = match item.response.status.as_str() {
+            "authorized" => common_enums::AttemptStatus::Authorized,
+            "processed" => common_enums::AttemptStatus::Charged,
+            "cancelled" | "voided" => common_enums::AttemptStatus::Voided,
+            "declined" => {
+                // Handle declined payments, including duplicate attempts
```

Comment:
this kind of mapping is not needed at all when we can directly make use of the existing enums